### PR TITLE
check if nspecies>0 before parsing particle_shape

### PR DESCRIPTION
**Pull request by RevathiJambunathan**
New input parameter `algo.particle_shape` is mandatory only if there is at least one particle species or one laser species.

### CODE REVIEW

**Comment by EZoni** [comment]
Thank you very much for the fix, Reva! I think we might need to move also the block (right below the one you already modified)
```cpp
        if ((maxLevel() > 0) && (particle_shape > 1) && (do_pml_j_damping == 1))
        {
            amrex::Warning("\nWARNING: When algo.particle_shape > 1,"
                           " some numerical artifact will be present at the interface between coarse and fine patch."
                           "\nWe recommend setting algo.particle_shape = 1 in order to avoid this issue");
        }
```
inside the `if` condition on the number of species that you added, otherwise `particle_shape` is not defined outside that scope.

**Comment by RevathiJambunathan** [comment]
oh thank you for catching that!

File: Source/WarpX.cpp [hunk 1]
```diff
@@ -790,31 +790,37 @@ WarpX::ReadParameters ()
...
-        int particle_shape;
-        if (pp_algo.query("particle_shape", particle_shape) == false)
-        {
-            amrex::Abort("\nalgo.particle_shape must be set in the input file:"
-                         "\nplease set algo.particle_shape to 1, 2, or 3");
-        }
-        else
-        {
-            if (particle_shape < 1 || particle_shape > 3)
+        // check if any nspecies > 0 before testing entry for algo.particle_shape
```
Comment (EZoni) [inline Source/WarpX.cpp]: Sorry for the very minor detail, I can commit the small change directly, to speed up the merge. Just want to make sure that the inline comment is as clear as possible (we are both checking and parsing the entry for `algo.particle_shape` at the same time, so this is also the place where we possibly assign the particle shape order and not only test that the user entered something acceptable):
```suggestion
        // Parse algo.particle_shape and check that input is acceptable
        // (do this only if there is at least one particle species)
```

**Review by EZoni — APPROVED** [review]
Thank you so much for the PR! I suggested a very minor change (just clarifying the inline comment, see below). I will maybe commit it directly myself, to speed up the merge.

**Comment by RevathiJambunathan** [comment]
oops just committed :)

File: Source/WarpX.cpp (hunk 1 above)
Comment (RevathiJambunathan) [reply Source/WarpX.cpp]: thank you @EZoni

**Comment by EZoni** [comment]
Actually I saw that the test `embedded_boundary_cube` seemed to have failed before. Now with your fix I think we might need to remove the input `algo.particle_shape` from the input files of the CI tests that have no particles, because we run CI with `amrex.abort_on_unused_inputs=1`. Does this make sense?

**Comment by RevathiJambunathan** [comment]
> Actually I saw that the test `embedded_boundary_cube` seemed to have failed before. Now with your fix I think we might need to remove the input `algo.particle_shape` from the input files of the CI tests that have no particles, because we run CI with `amrex.abort_on_unused_inputs=1`. Does this make sense?

Great!!!! I was just about to investigate why it would fail! you beat me to it! Thank you @EZoni ! :) 
Yes, I will remove it from there.

**Comment by EZoni** [comment]
Thanks a lot, Reva!

Sorry for the extra request, but could you maybe update the documentation of `algo.particle_shape` as well?

When we say
_"Note that this input parameter is not optional and must always be set in all input files."_
we could replace it with
_"Note that this input parameter is not optional and must always be set in all input files (provided that there is at least one particle species in the simulation)."_

Otherwise I'm afraid users will set it in their simulation no matter what, then get the warning of unused input parameter when they run the code, and think that the documentation is wrong or misleading. What do you think? Thanks a lot in advance!

**Comment by RevathiJambunathan** [comment]
Thank you so much for looking into this detail, Edoardo.
This commit adds a clarification in the documentation. 72850bf

File: Source/WarpX.cpp [hunk 2]
```diff
@@ -790,31 +790,43 @@ WarpX::ReadParameters ()
...
-        if (pp_algo.query("particle_shape", particle_shape) == false)
-        {
-            amrex::Abort("\nalgo.particle_shape must be set in the input file:"
-                         "\nplease set algo.particle_shape to 1, 2, or 3");
-        }
-        else
-        {
-            if (particle_shape < 1 || particle_shape > 3)
+        // Parse algo.particle_shape and check that input is acceptable
+        // (do this only if there is at least one particle species)
```
Comment (EZoni) [inline Source/WarpX.cpp]: I know I'm annoying :sweat_smile::
```suggestion
        // (do this only if there is at least one particle or laser species)
```

Comment (RevathiJambunathan) [reply Source/WarpX.cpp]: Thank you Edoardo!

**Review by EZoni — APPROVED** [review]
Looks great, thank you for the latest fixes!

**Comment by RevathiJambunathan** [comment]
Thank you for your suggestions, Edoardo :)

**Pull request merged**